agent/node.go

```diff
@@ -616,12 +616,14 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			go n.initManagerConnection(ctx, ready)
 
 			go func() {
-				select {
-				case <-ready:
-				case <-ctx.Done():
-				}
-				if ctx.Err() == nil {
-					n.remotes.Observe(api.Peer{NodeID: n.nodeID, Addr: n.config.ListenRemoteAPI}, 5)
+				// this happens only on initial start
+				if ready != nil {
```

Comment:
I think this can be outside the goroutine. If should be safe to set the ready nil outside the goroutine as well because atm this goroutine could still be executing on next for cycle.